bee-network/src/swarm/protocols/gossip/behavior.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{
+    event::{GossipEvent, GossipEventBuilder},
+    handler::GossipHandler,
+};
+
+use crate::network::meta::{ConnectionInfo, Origin};
+
+use libp2p::{
+    core::{connection::ConnectionId, ConnectedPoint},
+    swarm::{NegotiatedSubstream, NetworkBehaviour, NetworkBehaviourAction, ProtocolsHandler},
+    PeerId,
+};
+use log::trace;
+
+use std::{
+    collections::{HashMap, VecDeque},
+    sync::atomic::{AtomicBool, Ordering},
+    task::Poll,
+};
+
+pub static GOSSIP_ORIGIN: AtomicBool = AtomicBool::new(false);
+
+#[derive(Eq, PartialEq, Hash, Debug, Clone, Copy)]
+struct Id(PeerId, ConnectionId);
+
+#[derive(Default)]
+pub struct Gossip {
+    builders: HashMap<PeerId, GossipEventBuilder>,
+    // events produced by gossip handlers
+    events: VecDeque<GossipEvent>,
+}
+
+impl Gossip {
+    pub fn new() -> Self {
+        Self::default()
+    }
+}
+
+impl NetworkBehaviour for Gossip {
+    type ProtocolsHandler = GossipHandler;
+    type OutEvent = GossipEvent;
+
+    fn new_handler(&mut self) -> Self::ProtocolsHandler {
+        // FIXME
```

Comment:
I'd add a description explaning what to fix so someone else can do it

Reply:
you're right. I am trying to come up with a concise description of the issue here, but it's not that easy.

Reply:
I added documentation that tries to describe the issue, and why it needs fixing although it's not a severe issue. I put that explanation above the `AtomicBool` `GOSSIP_ORIGIN` in `gossip::behavior`.